3-networks/modules/restricted_shared_vpc/nat.tf

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  NAT Cloud Router & NAT config
+ *****************************************/
+
+resource "google_compute_router" "nat_router_region1" {
+  count   = var.nat_enabled ? 1 : 0
+  name    = "cr-${local.vpc_name}-${var.default_region1}-nat-router"
+  project = var.project_id
+  region  = var.default_region1
+  network = module.main.network_self_link
+
+  bgp {
+    asn = var.nat_bgp_asn_region1
```

Comment:
If possible, we should use a single ASN for both NAT Cloud Routers to limit the number of inputs required.